app/views/pets/show.html.erb

```diff
@@ -2,7 +2,11 @@
 
 <div class="pets">
 <section class="pets-<%= @pet.id %>">
-  <button><%= link_to 'Favorite', "/favorites/#{@pet.id}", method: :patch %></button>
+  <% if favorite.contents[@pet.id.to_s].nil? %>
```

Comment:
Not sure why `<% if favorite.contents[@pet.id.to_s]==0 %>` resulted in a bunch of errors.
Made sense to me: if this value of favorited pet is 0, show the favorite button.